resource_customizations/core.spinkube.dev/SpinApp/health.lua

```diff
@@ -0,0 +1,56 @@
+hs = {}
+
+-- Check if status exists
+if not obj.status then
+  hs.status = "Degraded"
+  hs.message = "No status available"
+  return hs
+end
+
+-- Initialize variables for conditions
+local available = false
+local progressing = false
+local availableMessage = ""
+local progressingMessage = ""
+
+-- Check conditions
+if obj.status.conditions then
+  for _, condition in ipairs(obj.status.conditions) do
+    if condition.type == "Available" then
+      available = condition.status == "True"
+      availableMessage = condition.message or "Application availability status"
+    elseif condition.type == "Progressing" then
+      progressing = condition.status == "True"
+      progressingMessage = condition.message or "Application progress status"
+    end
```

Comment:
Looks like we have a "last one wins" situation. Do we wanna mark progressing if _any_ progressing condition is True, regardless of the state of any Available conditions?

Reply:
reordered the flow and improved checking conditions - thanks!